src/utils/parse_utils.ts

```diff
@@ -10,7 +11,115 @@ import {
+    includedSources?: string;
+    intentOnFilling?: string;
+    apiKey?: string;
+}
+
+/**
+ * This constant contains, as keys, all ERC20BridgeSource types except from `Native`.
+ * As we add more bridge sources to AssetSwapper, we want to keep ourselves accountable to add
+ * them to this constant. Since there isn't a good way to enumerate over enums, we use a obect type.
+ * The type has been defined in a way that the code won't compile if a new ERC20BridgeSource is added.
+ */
+const ALL_EXCEPT_NATIVE: { [key in Exclude<ERC20BridgeSource, ERC20BridgeSource.Native>]: boolean } = {
+    Uniswap: true,
+    Balancer: true,
+    Curve: true,
+    Eth2Dai: true,
+    Kyber: true,
+    LiquidityProvider: true,
+    MultiBridge: true,
+    Uniswap_V2: true,
+};
+
 export const parseUtils = {
+    parseRequestForExcludedSources(
+        request: ParseRequestForExcludedSourcesParams,
+        validApiKeys: string[],
+        endpoint: 'price' | 'quote',
+    ): { excludedSources: ERC20BridgeSource[]; nativeExclusivelyRFQT: boolean } {
+        // Ensure that both filtering arguments cannot be present.
+        if (request.excludedSources !== undefined && request.includedSources !== undefined) {
+            throw new ValidationError([
+                {
+                    field: 'excludedSources',
+                    code: ValidationErrorCodes.IncorrectFormat,
+                    reason: ValidationErrorReasons.ConflictingFilteringArguments,
+                },
+                {
+                    field: 'includedSources',
+                    code: ValidationErrorCodes.IncorrectFormat,
+                    reason: ValidationErrorReasons.ConflictingFilteringArguments,
+                },
+            ]);
+        }
+
+        // If excludedSources is present, parse the string array and return
+        if (request.excludedSources !== undefined) {
+            return {
+                excludedSources: parseUtils.parseStringArrForERC20BridgeSources(request.excludedSources.split(',')),
+                nativeExclusivelyRFQT: false,
+            };
+        }
+
+        if (request.includedSources !== undefined) {
+            // Only RFQT is eligible as of now
+            if (request.includedSources === 'RFQT') {
+                // We assume that if a `takerAddress` key is present, it's value was already validated by the JSON
+                // schema.
+                if (request.takerAddress === undefined || request.takerAddress.length === 0) {
+                    throw new ValidationError([
+                        {
+                            field: 'takerAddress',
+                            code: ValidationErrorCodes.FieldInvalid,
+                            reason: ValidationErrorReasons.TakerAddressInvalid,
+                        },
+                    ]);
+                }
+
+                // We enforce a valid API key - we don't want to fail silently.
+                if (!validApiKeys.includes(request.apiKey)) {
```

Comment:
Should we have two separate errors: one for when an apiKey is not specified (RequiredField) and one for when it is specified but invalid? (FieldInvalid+ InvalidApiKey)